localstack/services/sns/provider.py

```diff
@@ -104,10 +103,12 @@ def get_moto_backend(account_id: str, region_name: str) -> SNSBackend:
         return sns_backends[account_id][region_name]
 
     @staticmethod
-    def _get_topic(arn: str) -> Topic:
-        arn_data = parse_arn(arn)
-        backend = SnsProvider.get_moto_backend(arn_data["account"], arn_data["region"])
-        return backend.get_topic(arn)
+    def _get_topic(arn: str, region: str) -> Topic:
```

Comment:
It can make sense, right. We would need to try it against AWS to see if the exception raised is simply `Topic not found`, or if there's a specific exception for cross region access? In the meantime, I can simply pass the context, very good idea, thanks!